### DISCUSSION TITLE
Future of Scaffolder

### DISCUSSION THREAD

**Discussion by dl-tg** [dl-tg/scaffolder#9] [General]
It seems, atleast for me, that the tool is pretty much complete and there's really nothing to add. However, if you've got an idea, write it here 👇🏻

**Comment by Gwali-1** [comment]
Same here , it serves a purpose and does it well as far as I'm concerned. Further improvements in my opinion will be around making its usability more streamlined eg at some point I felt there were a lot of command flags to provide and was thinking of a way to make it easy to set up your directory structure in a more intuitive way . Also the idea of having a scaffold yaml file market is something I thought of and is not a bad idea but then I thought how to structure projects is different for each person,  but would be nice to you know be able to just build your project structure with some standard config you didn't write yourself .

**Comment by dl-tg** [comment]
> Same here , it serves a purpose and does it well as far as I'm concerned. Further improvements in my opinion will be around making its usability more streamlined eg at some point I felt there were a lot of command flags to provide and was thinking of a way to make it easy to set up your directory structure in a more intuitive way . Also the idea of having a scaffold yaml file market is something I thought of and is not a bad idea but then I thought how to structure projects is different for each person, but would be nice to you know be able to just build your project structure with some standard config you didn't write yourself .

@Gwali-1 Few languages have strictly defined a structure. Most of the time it's up to yourself and your preferences. This is why it's hard to make a standard confing, because only some frameworks like JavaScript ones have their structure, but it's already handled by their CLI tools (CRA, Vite).

I've made a repository with some configs for reference though.

**Reply by Gwali-1** [reply]
thats true.

**Comment by massenz** [comment]
This looks really useful - it would also be a nice complement to my [`massenz/common-utils`](https://github.com/massenz/common-utils) collection of utilities, but I couldn't gather (reading the usage instructions) whether it would be possible to *include* an external file, as opposed to having to add its *contents* inside the YAML.

In other words, instead of:
```
src:
  Makefile: |
      ... now a 100+ lines of makefile... (with all the complexity of tabs indentation etc)
```
having something like:
```
src:
  include:
    - name: Makefile
      content: ~/templates/Makefile
    - name: build.settings
      content: ~/templates/build/settings.template
```
and this would just copy the files into the appropriate places (bonus points if it could also do Go template var substitution).
